### DISCUSSION TITLE
Display cover of radio stations

### DISCUSSION THREAD

**Discussion by payback007** [jcorporation/myMPD#557] [Q&A] [answered]
Hi,

thanks for the great work on mympd! I really enjoy this cool piece of software!

For playback actual radio stations, I'm sending an adtitional URI directly to the mpd-client, this is also shown/controlled by mympd after playback started. In previous versions the cover of the radio station was also displayed by mympd, in the new version 8.0.4 there is no cover displayed anymore.

**Comment by jcorporation** [comment]
In my environment the display of coverimages for streams works the same as in older versions. I need debug output to investigate further.

**Comment by payback007** [comment]
I did also some investigation of the log files and in the meantime following:
- mympd tries to download the cover from the path of the radio station, for example I tried this station: http://br-br3-live.cast.addradio.de/br/br3/live/mp3/mid
- no cover could be found there: 

`Aug 29 17:48:06 multiroom mympd[21055]: INFO     webserver /tmp/myMPD/src/web_server.c:330: HTTP request (5): GET /albumart/http://br-br3-live.cast.addradio.de/br/br3/live/mp3/mid`
`Aug 29 17:48:06 multiroom mympd[21055]: DEBUG    webserver /tmp/myMPD/src/web_server/web_server_albumart.c:85: Handle albumart for uri "/albumart/http://br-br3-live.cast.addradio.de/br/br3/live/mp3/mid"`
`Aug 29 17:48:06 multiroom mympd[21055]: DEBUG    webserver /tmp/myMPD/src/web_server/web_server_albumart.c:98: Check for stream cover /var/lib/mympd1/pics/br-br3-live_cast_addradio_de_br_br3_live_mp3_mid`
`Aug 29 17:48:06 multiroom mympd[21055]: DEBUG    webserver /tmp/myMPD/src/web_server/web_server_utility.c:175: Serving file "/assets/coverimage-stream.svg" ()`

- the cover of the radio station is also not displayed in webbrower if I use direct link -> so I think, it is not a topic from mympd

Work arround at the moment:
- local storage of the cover for each radio station in "/var/lib/mympd1/pics/"
- name of the file is the same as the web address
- file format is *.png
- cover is displayed / changed if the radio station is playing

-> result can be found here: [result_working_radio_cover.txt](https://github.com/jcorporation/myMPD/files/7072871/result_working_radio_cover.txt)


Problem still existing:
- no cover path provided by mympd for "albumart" through API call

**Comment by jcorporation** [comment]
It seems all is working as designed.
Stream: http://br-br3-live.cast.addradio.de/br/br3/live/mp3/mid

Coverfile must be: /var/lib/mympd1/pics/br-br3-live_cast_addradio_de_br_br3_live_mp3_mid

with extension: jpg, png, webp, etc.

There is no api call for this, use /albumart/http://br-br3-live.cast.addradio.de/br/br3/live/mp3/mid to get the cover.

**Comment by payback007** [comment]
Yes for sure, but in previous version the path was delivered back by api call... Why is the current path not given back by api call? I only get this back:

**http://IP:PORT/albumart/**  but the part of http-radio-string is missinng.

manually editing the full path is working at all **http://IP:PORT/albumart/http://br-br3-live.cast.addradio.de/br/br3/live/mp3/mid**

**Comment by jcorporation** [comment]
Which api call you mean?

There is no api call because the myMPD web ui does not need an api call for this.

**Comment by payback007** [comment]
Overall I use mpd/mympd in a smarthome multiroom application running 4 instances in parallel for different users/rooms. Every person in the household has it's own mympd/mpd package. Aditional in smarthome interface rooms, valume, etc. can be controlled via api without need going directly to mympd-

ThereforeI control the music playback in two different ways:
1) mympd for album / audiobooks from server database and so on with a drop-in webinterface in external smarthonme control software
2) external radio station list in webinterface with direct input to MPD -> with every change of the filename (radio, database, doesn't matter from what change), api of mympd is called to refresh the actual album cover with following string:

`URI=$(curl -s -k http://IP:PORT/api/ -d'{"jsonrpc":"2.0","id":0,"method":"MYMPD_API_PLAYER_CURRENT_SONG","params":{}}' | jq -r ".result.images | .[]")`

This is working very reliable for files in the database. For the radio station, the api call remains empty, the path is not fully delivered any more. In previous version this was working also very well for the radio stations.

**Reply by jcorporation** [reply]
Ok, I can not remember any change. In which version this was working?

With current version you can take the Song Uri from above call and prefix this with /albumart/ and you get the cover for songs in the database and streams.

```
URI=$(curl -s -k http://IP:PORT/api/ -d'{"jsonrpc":"2.0","id":0,"method":"MYMPD_API_PLAYER_CURRENT_SONG","params":{}}' | jq -r ".result.uri")
URI="/albumart${URI}"
```

**Reply by payback007** [reply]
Thank you, it is now working again as in version 7!

Sorry for the confusion, I thought I would need a *.png / *.jpg file path to show the actual cover!

**Comment by jcorporation** [comment, marked as answer]
https://github.com/jcorporation/myMPD/discussions/557#discussioncomment-1250661
